kube/src/api/subresource.rs

```diff
@@ -208,3 +211,177 @@ where
         Ok(self.client.request_text_stream(req).await?)
     }
 }
+
+// ----------------------------------------------------------------------------
+// Attach subresource
+// ----------------------------------------------------------------------------
+/// Params for attaching
+///
+/// Note that the server rejects when none of `stdin`, `stdout`, `stderr` are attached.
+#[cfg(feature = "ws")]
+pub struct AttachParams {
+    /// The container in which to execute the command.
+    /// Defaults to only container if there is only one container in the pod.
+    pub container: Option<String>,
+    /// If `true`, the contents will be redirected to the standard input stream of the pod. Defaults to `false`.
+    pub stdin: bool,
+    /// If `true`, the standard out stream of the pod will be redirected to it. Defaults to `true`.
+    pub stdout: bool,
+    /// If `true`, the standard error stream of the pod will be redirected to it. Defaults to `true`.
+    pub stderr: bool,
+    /// If `true`, TTY will be allocated for the attach call. Defaults to `false`.
+    pub tty: bool,
+}
+
+#[cfg(feature = "ws")]
+impl Default for AttachParams {
+    // Default matching the server's defaults.
+    fn default() -> Self {
+        Self {
+            container: None,
+            stdin: false,
+            stdout: true,
+            stderr: true,
+            tty: false,
+        }
+    }
+}
+
+#[cfg(feature = "ws")]
+impl Resource {
+    /// Attach to a pod
+    pub fn attach(&self, name: &str, ap: &AttachParams) -> Result<http::Request<()>> {
+        let base_url = self.make_url() + "/" + name + "/" + "attach?";
+        let mut qp = url::form_urlencoded::Serializer::new(base_url);
+
+        if ap.stdin {
+            qp.append_pair("stdin", "true");
+        }
+        if ap.stdout {
+            qp.append_pair("stdout", "true");
+        }
+        if ap.stderr {
+            qp.append_pair("stderr", "true");
+        }
+        if ap.tty {
+            qp.append_pair("tty", "true");
+        }
+        if let Some(container) = &ap.container {
+            qp.append_pair("container", &container);
+        }
+
+        let req = http::Request::get(qp.finish());
+        req.body(()).map_err(Error::HttpError)
+    }
+}
+
+/// Marker trait for objects that has attach
+#[cfg(feature = "ws")]
+pub trait AttachingObject {}
```

Comment:
Maybe `AttachableObject` is better?

Reply:
Ah, here the convention does break a bit. I think you're right. Maybe we should make the convention of these trait names to "reflect the property implementing it gives us". So `Attachable` would be better.

If that's the convention, we also technically don't break `LoggingObject` (which arguably should have been `LoggableObject`)